### PR TITLE
fixed: use the dyn url for protocol check if available

### PR DESCRIPTION
**Pull request by notspiff**
broke inputstream.rtmp usage.

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDInputStreams/InputStreamAddon.cpp [hunk 1]
```diff
@@ -82,7 +82,12 @@ bool CInputStreamAddon::Supports(BinaryAddonBasePtr& addonBase, const CFileItem
     return (addon.asString() == addonBase->ID());
 
   // check protocols
-  std::string protocol = fileitem.GetURL().GetProtocol();
+  std::string protocol;
+  if (!fileitem.GetDynPath().empty())
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDInputStreams/InputStreamAddon.cpp]: no need to test for empty. GetDynPath already  returns GetPath if dynPath is empty

**Comment by notspiff** [comment]
rebased.
